File: Patching Array.go
func minPatches(nums []int, n int) int {
    var i, ans, total int
    m := len(nums)
    j := 1
    for j <= n {
        for i < m && nums[i] <= j {
            total += nums[i]
            i++
        }

        if total < j {
            ans++
            total += j
        }
        j = total+1
    }
    return ans
}

File: Patching Array.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        i = 0
        total = 0
        m = len(nums)
        ans = 0
        j = 1
        while j <= n:
            while i < m and nums[i] <= j:
                total += nums[i]
                i += 1

            # print(total, j)
            if total < j:
                ans += 1
                total += j
            j = total+1

        return ans
